addons/web_unsplash/static/src/components/media_dialog/image_selector.js

```diff
@@ -126,7 +126,10 @@ patch(ImageSelector.prototype, 'image_selector_unsplash', {
             const { isMaxed, images } = await this.unsplash.getImages(this.state.needle, offset, this.NUMBER_OF_RECORDS_TO_DISPLAY, this.props.orientation);
             this.state.isFetchingUnsplash = false;
             this.state.unsplashError = false;
-            const records = images.map(record => {
+            // Ignore duplicates.
+            const existingIds = this.state.unsplashRecords.map(existing => existing.id);
```

Comment:
Wouldn't it be possible to get twice the same image in a single batch? I guess it can't happen, but I would have thought the same of fetching the same image in subsequent fetches, so...
Do you know in which cases Unsplash can send an image already fetched?

Reply:
My guess is that there is no cursor kept in the session involved when we:
- fetch 30 from start
- fetch 30 from offset 30
- fetch 30 from offset 60
- ...

And therefore, the sort order of results returned by Unsplash is not stable across successive calls - it might typically involve information such as "how many times was a picture returned", "when was a picture returned last"... which change with every call that is made.